### PR TITLE
feat: add `BotIntegration.scopes`

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

see https://github.com/discord/discord-api-docs/pull/5311

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/integrations.py [hunk 1]
```diff
@@ -409,13 +409,18 @@ class BotIntegration(Integration):
         The integration account information.
     application: :class:`IntegrationApplication`
         The application tied to this integration.
+    scopes: List[:class:`str`]
```
Comment (onerandomusername) [inline disnake/integrations.py:412]: Would it make sense to turn this into a set?

Comment (shiftinv) [reply disnake/integrations.py]: Sure, sounds good.
https://github.com/DisnakeDev/disnake/pull/729/commits/1c3be76b2d6c3a20692301dc6d1c36321f0e7127

Comment (onerandomusername) [reply disnake/integrations.py]: That was a question, I wasn't sure it was a good idea! 😅

**Comment by onerandomusername** [comment]
I think this is a good attribute to add to the repr of bot integrations.

File: disnake/integrations.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/integrations.py:412]: Additionally, we use Lists elsewhere for scopes already so we should keep it a lsit.

**Comment by shiftinv** [comment]
> I think this is a good attribute to add to the repr of bot integrations.

Done in https://github.com/DisnakeDev/disnake/pull/729/commits/1a31521e4ae1e7ea40ed011c0c06474966021281.

**Review by onerandomusername — APPROVED** [review]

**Pull request merged**